c10/util/BFloat16-math.h

```diff
@@ -68,6 +68,12 @@ template <
 inline T expm1(T a) {
   return std::expm1(float(a));
 }
+template <
+    typename T,
+    typename std::enable_if_t<is_reduced_floating_point_v<T>, int> = 0>
+inline bool isfinite(T a) {
+  return std::isfinite(float(a));
```

Comment:
How is this tested?
I don't know the exact bf16 specs but the conversion to float is going to properly preserve everything we need?

Reply:
We do have extensive test for 16-bit types in https://github.com/pytorch/pytorch/blob/main/c10/test/util/bfloat16_test.cpp and https://github.com/pytorch/pytorch/blob/main/c10/test/util/Half_test.cpp